### PR TITLE
Feature: Add load_CB_speeches function

### PR DESCRIPTION
**Pull request by robin-mader-bis**
This Pull Request adds a `load_CB_speeches` function for downloading the soon to be published Central Bankers Speeches dataset from www.bis.org.

Summary of changes:
- Add `load_CB_speeches` function and related documentation
- Add `internals` module, intended for functions which are intended for package internal use only
- Add `settings` module, for package internal setting constants
- Add citation for Central Bankers speeches dataset

### CODE REVIEW

File: 00_datasets.qmd [hunk 1]
```diff
@@ -64,6 +64,35 @@ X.head()
...
+<!-- load_CB_speeches -->
+```{python}
+#| output: asis
+#| echo: false
+show_doc(load_CB_speeches)
+```
+
+This function downloads the Central bankers speeches dataset from the
+BIS website (www.bis.org). More information on the dataset can be found
+on the BIS website.
```
Comment (dkgaraujo) [inline 00_datasets.qmd:76]: Please include the speeches URL after "on the BIS website"

File: gingado/datasets.py [hunk 2]
```diff
@@ -32,6 +38,92 @@ def load_BarroLee_1994(
...
+            Path(cb_speeches_file_path).parent.mkdir(exist_ok=True)  # Ensure parent dir exists
+            cb_speeches_year_df.to_csv(timestamped_file_path, index=False)
+
+        # Verify that the file in the cache is valid
+        try:
+            timestamped_file_path = get_latest_timestamped_file_path(cb_speeches_file_path)
+            verify_df = pd.read_csv(timestamped_file_path)
+            assert len(verify_df) > 0, f'CB speeches dataset at {timestamped_file_path} is empty.'
+        except Exception as ex:
+            raise RuntimeError('Verification error. See previous exception.') from ex
```
Comment (dkgaraujo) [inline gingado/datasets.py:118]: Is it possible to refer to previous exceptions?

File: gingado/datasets.py [hunk 3]
```diff
@@ -32,6 +38,92 @@ def load_BarroLee_1994(
...
+            filename_no_extension = (
+                CB_SPEECHES_ZIP_BASE_FILENAME if y == 'all' else f'{CB_SPEECHES_ZIP_BASE_FILENAME}_{y}'
+            )
+            zip_url = CB_SPEECHES_BASE_URL + filename_no_extension + '.zip'
+
+            # Download the zip file, unzip it and parse the CSV file:
+            with requests.get(zip_url, timeout=timeout) as response:
+                zip_file_content = response.content
+                speeches_zip = ZipFile(BytesIO(zip_file_content))
+                speeches_zip_file = speeches_zip.open(filename_no_extension + '.csv', 'r')
```
Comment (dkgaraujo) [inline gingado/datasets.py:104]: shouldn't this also be within a context, ie `with speeches_zip.open()...:` ?

**Comment by dkgaraujo** [comment]
Many thanks @robin-mader-bis. The only thing I'm missing is the new rendered datasets.html. It's a matter of simply rendering the corresponding qmd file: `quarto render 00_datasets.qmd`. Please include also the resulting docs\datasets.html file in the PR. The rest shoul;d be good to go, pending also the comments above.

**Comment by robin-mader-bis** [comment]
@dkgaraujo My bad, I thought this would be added automatically by the CI. Should I also include other files that are changed during the render process, or only the docs\datasets.html?

File: gingado/datasets.py (hunk 2 above)
Comment (robin-mader-bis) [reply gingado/datasets.py:118]: The `from ex` clause should ensure we get both exceptions in approx. the following format:
\<original exception details and stacktrace\>
During the handling of the previous exception, another exception occured:
\<our exception details and stacktrace\>

**Comment by dkgaraujo** [comment]
> @dkgaraujo My bad, I thought this would be added automatically by the CI. Should I also include other files that are changed during the render process, or only the docs\datasets.html?

No problem. The CI part will be added later, but for now the PRs need to render all the files automatically. 

Since the only "numbered" quarto file you changed was `00_datasets.qmd`, there is no need to re-render the whole project, just that file.

File: 00_datasets.qmd [hunk 4]
```diff
@@ -64,6 +64,35 @@ X.head()
...
 
+<!-- load_CB_speeches -->
+```{python}
+#| output: asis
+#| echo: false
+show_doc(load_CB_speeches)
+```
+
+This function downloads the Central bankers speeches dataset from the
+BIS website (www.bis.org). More information on the dataset can be found
```
Comment (dkgaraujo) [inline 00_datasets.qmd:75]: Please also include the bibtex citation (similar to how is done above for the Barro-Lee dataset)

File: 00_datasets.qmd (hunk 1 above)
Comment (robin-mader-bis) [reply 00_datasets.qmd]: Link has been a dded

File: gingado/datasets.py (hunk 3 above)
Comment (robin-mader-bis) [reply gingado/datasets.py]: You are right. I added the corresponding context.

File: 00_datasets.qmd (hunk 4 above)
Comment (robin-mader-bis) [reply 00_datasets.qmd]: Citation has been added.

**Comment by dkgaraujo** [comment]
One more thing I uncovered testing on python 3.9: we need to add the folling import at the very beginning of `datasets.py`, similar to `estimators.py`:

```
from __future__ import annotations  # Allows forward annotations in Python < 3.10
```

This allows for arguments that have multiple types.

**Comment by robin-mader-bis** [comment]
@dkgaraujo Good catch, I added the import as suggested

**Review by dkgaraujo — APPROVED** [review]

**Pull request merged**